http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcTest.java

```diff
@@ -179,12 +222,30 @@ private void performAuthentication(InputStream oidcConfig, String username, Stri
             TestingHttpServerResponse response = request.getResponse();
             assertEquals(loginToKeycloak ? HttpStatus.SC_MOVED_TEMPORARILY : HttpStatus.SC_FORBIDDEN, response.getStatusCode());
             assertEquals(Status.NO_AUTH, request.getResult());
+            if (expectedScope != null){
+                assertTrue(response.getFirstResponseHeaderValue("Location").contains("scope=" + expectedScope));
+            }
 
             if (loginToKeycloak) {
                 client.setDispatcher(createAppResponse(mechanism, expectedDispatcherStatusCode, expectedLocation, clientPageText));
-                TextPage page = loginToKeycloak(username, password, requestUri, response.getLocation(),
-                        response.getCookies()).click();
-                assertTrue(page.getContent().contains(clientPageText));
+
+                if (checkInvalidScopeError){
+                    WebClient webClient = getWebClient();
+                    List<HttpServerCookie> cookies = response.getCookies();
+
+                    if (cookies != null) {
+                        for (HttpServerCookie cookie : cookies) {
+                            webClient.addCookie(getCookieString(cookie), requestUri.toURL(), null);
+                        }
+                    }
```

Comment:
I am not sure. This is what the function was doing where I took the code from. 
The reason why I copied the code over instead of using the function is because that other function was expecting an HTMLPage and I am expecting a TextPage here

Reply:
Implemented changes except this.

Reply:
I think it can be removed here.